splinepy/io/svg.py

```diff
@@ -0,0 +1,940 @@
+
+            Parameters:
+            png_tag : bytes
+              PNG tag identifier.
+            data : bytes
+              Data to be packed.
+
+            Returns
+            -------
+            bytes
+              Packed PNG chunk.
+            """
+            chunk_head = png_tag + data  # Concatenate the tag and data
+            return (
+                # Pack the length of chunk
+                struct.pack("!I", len(data))
+                +
+                # Concatenate the chunk header
+                chunk_head
+                +
+                # Pack CRC-32 checksum
+                struct.pack("!I", 0xFFFFFFFF & zlib.crc32(chunk_head))
+            )
+
+        pure_data = b"".join(
+            [
+                b"\x89PNG\r\n\x1a\n",  # PNG file signature
+                png_pack(
+                    b"IHDR", struct.pack("!2I5B", width, height, 8, 6, 0, 0, 0)
+                ),  # PNG header chunk
+                # PNG data chunk (compressed)
+                png_pack(b"IDAT", zlib.compress(raw_data, 9)),
+                png_pack(b"IEND", b""),  # PNG end chunk
+            ]
+        )
+
+        # Return encoded
+        if as_base64:
+            return base64.b64encode(pure_data)
+        else:
+            return pure_data
+
+    # Sample the spline
+    resolution = spline.show_options.get("resolutions", 100)
+    sampled_spline = _sample_spline(spline, resolution)
+    data = spline.show_options.get("data", None)
+
+    # Check
+    if data is None:
+        raise ValueError(
+            "There is no data provided, although data plot is requested"
+        )
+
+    # Process the scalar field
+    _process_scalar_field(spline, data, sampled_spline, res=resolution)
+    # scalar bar distorts the dimensions of spline bitmap image
+    sampled_spline.show_options["scalarbar"] = False
+    # Set a resolution (This needs to be improved still)
+    pixel_size = (resolution * 10, resolution * 10)
+
+    # Create a Plotter
+    plotter = _Plotter(
+        shape=(1, 1),  # Only one field
+        N=1,  # Number of things to plot
+        size=pixel_size,  # Resolution, i.e. pixel density
+        sharecam=True,
+        offscreen=True,
+        title="",
+        bg=(255, 255, 255),
+        axes=0,
+    )
+    plotter.show(
+        sampled_spline.showable(),
+    )
+
+    # Extract bounding box
+    x_min, y_min, x_max, y_max = sampled_spline.bounds().ravel()
+
+    # Extract the bitmap and transform to RGBA
+    bitmap = plotter.screenshot(asarray=True)
+
+    if bitmap.shape[2] == 3:
+        alpha_layer = (
+            np.ones((bitmap.shape[0], bitmap.shape[1]), dtype=bitmap.dtype)
+            * 255
+        )
+        alpha_layer[np.all(bitmap == 255, axis=-1)] = 0
+        bitmap = np.concatenate(
+            (bitmap, alpha_layer.reshape(*alpha_layer.shape, 1)), axis=-1
+        )
+
+    # Crop image
+    has_pixels = np.where(np.any(alpha_layer != 0, axis=1))[0]
```

Comment:
If you'd like to save more time, you can save 'alpha_layer != 0' to a variable and use it for both any

Reply:
done